<a href="https://colab.research.google.com/github/SahanaDil/Projects/blob/main/06Wisconsin_Breast_Cancer_MLModel_using_Gaussian_Naive_Bayes_Classifier.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Developing a ML Model using Gaussian Naive Bayes Classifer with Wisconsin Breast Cancer Dataset

#### The goal of this analysis is to develop a ML Model to accurately distinguish between a benign and malignant tumor to aid in clinical diagnosis.

In [ ]:
from datetime import date

today = date.today()
print(f"Today's date is {today}. its me Sahana.D !")


Today's date is 2025-02-13. its me Sahana.D !


In [1]:
from google.colab import drive

# This will prompt you to authorize access
drive.mount('/content/drive')


Mounted at /content/drive


In [2]:
#Data Aquisition
import pandas as pd
import numpy as np

df=pd.read_csv("/content/drive/MyDrive/wisc_bc_data.csv")

# DATA PRE PROCESSING
steps: 1) Understand data and its features
       2) Address the Missing values

In [3]:
df.shape #checking the overall shape

(569, 32)

###### my observation: There are 569 rows/observations and 32 columns/features in this dataset.

In [4]:
df.isnull().sum() #check the missing value

id                         0
diagnosis                  0
radius_mean                0
texture_mean               0
perimeter_mean             0
area_mean                  0
smoothness_mean            0
compactness_mean           0
concavity_mean             0
concave points_mean        0
symmetry_mean              0
fractal_dimension_mean     0
radius_se                  0
texture_se                 0
perimeter_se               0
area_se                    0
smoothness_se              0
compactness_se             0
concavity_se               0
concave points_se          0
symmetry_se                0
fractal_dimension_se       0
radius_worst               0
texture_worst              0
perimeter_worst            0
area_worst                 0
smoothness_worst           0
compactness_worst          0
concavity_worst            0
concave points_worst       0
symmetry_worst             0
fractal_dimension_worst    0
dtype: int64

###### my Observation : No Missing Values

In [5]:
df.info() #check datatypes

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 569 entries, 0 to 568
Data columns (total 32 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   id                       569 non-null    int64  
 1   diagnosis                569 non-null    object 
 2   radius_mean              569 non-null    float64
 3   texture_mean             569 non-null    float64
 4   perimeter_mean           569 non-null    float64
 5   area_mean                569 non-null    float64
 6   smoothness_mean          569 non-null    float64
 7   compactness_mean         569 non-null    float64
 8   concavity_mean           569 non-null    float64
 9   concave points_mean      569 non-null    float64
 10  symmetry_mean            569 non-null    float64
 11  fractal_dimension_mean   569 non-null    float64
 12  radius_se                569 non-null    float64
 13  texture_se               569 non-null    float64
 14  perimeter_se             5

In [ ]:
print(df.head(5)) #examine first 5 rows

print (df.tail(5))  #examine first 5 rows

         id diagnosis  radius_mean  texture_mean  perimeter_mean  area_mean  \
0    842302         M        17.99         10.38          122.80     1001.0   
1    842517         M        20.57         17.77          132.90     1326.0   
2  84300903         M        19.69         21.25          130.00     1203.0   
3  84348301         M        11.42         20.38           77.58      386.1   
4  84358402         M        20.29         14.34          135.10     1297.0   

   smoothness_mean  compactness_mean  concavity_mean  concave points_mean  \
0          0.11840           0.27760          0.3001              0.14710   
1          0.08474           0.07864          0.0869              0.07017   
2          0.10960           0.15990          0.1974              0.12790   
3          0.14250           0.28390          0.2414              0.10520   
4          0.10030           0.13280          0.1980              0.10430   

   ...  radius_worst  texture_worst  perimeter_worst  area_wor



##### my Observation:  Most features are of type float64. **The** id column is of type int64, and the target variable (diagnosis) is of type object.

##### ID column: we dont need it as its just incremental numbers hence we can drop it.
#####diagnosis is going to be the target variable. Diagnosis is categorical and has the values M and B (Malignant and Benign)

##### 30 Numerical features/columns are derived from cell nucleus charcateristics , such as radius_mean,texture_mean and area_mean etc

#EXPLORATORY DATA ANALYSIS